source/grid/tria_accessor.cc

```diff
@@ -1531,6 +1531,17 @@ CellAccessor<dim, spacedim>::set_direction_flag (const bool new_direction_flag)
 
 template <int dim, int spacedim>
 void
+CellAccessor<dim, spacedim>::set_active_cell_index (const unsigned int active_cell_index)
+{
+  Assert (this->used(), TriaAccessorExceptions::ExcCellNotUsed());
```

Comment:
In fact I'm going to change the code so that it's not even necessary any more that the cell is used and reset the flag also for unused ones.